routes/mylibrary.js

```diff
@@ -97,52 +100,99 @@ function addUserBook(userId, bookId, conditionId, callback)
+                        FROM Books
+                        INNER JOIN UserBooks ON UserBooks.bookId = Books.id
+                        INNER JOIN Transactions ON Transactions.userBookId = UserBooks.id 
+                        INNER JOIN TransactionStatusDates ON TransactionStatusDates.transactionId = Transactions.id
+                        INNER JOIN Conditions on Conditions.id = UserBooks.conditionId
+                        WHERE Transactions.requestorId = ? AND Transactions.statusId = 8 AND TransactionStatusDates.statusId = 4`;
     sql.pool.query(selectAllBooks, [userId], (err, result) => {
         if (err) {
             req.flash('error', 'Error retrieving all books. Try refreshing your page.')
             res.render('myLibrary')
         }
+        var copy;
+        var count = 0;
         for (let i = 0; i < result.length; i++) {
-            payload.library.push(new Book(result[i].userBookId, result[i].bookId, result[i].swap, result[i].title, result[i].imgUrl));
-            if (result[i].swap == 0) {
-                payload.rcvd++;
-            } else {
+            var data = {
+                userBookId: [result[i].userBookId],
+                swap: result[i].swap,
+                title: result[i].title,
+                author: result[i].author,
+                imgUrl: result[i].imgUrl,
+                condition: [result[i].description],
+                date: [result[i].date],
+                copies: 0
+            }
+            if (result[i].swap == 1) {
                 payload.avail++;
             }
+            if (i != 0 && copy.title == data.title){
+                if (count == 0){
+                    copy.copies++;
+                }
+                copy.userBookId.push(result[i].userBookId);
+                copy.condition.push(result[i].description);
+                copy.date.push(result[i].date);
+                copy.copies++;
+                count++;
+                continue;
+            }
+            payload.library.push(new Book(data));
+            copy = payload.library[payload.library.length-1];
         }
-        sql.pool.query(selectAllTitles, [], (err, result) => {
+        // all books rcvd in swaps
+        sql.pool.query(selectRCVDBooks, [userId], (err, result) => {
             if (err) {
-                req.flash('error', 'Error retrieving book by title.')
-                res.render('myLibrary', payload)
+                req.flash('error', 'Error retrieving all books. Try refreshing your page.')
+                res.render('myLibrary')
             }
+            var copy;
+            var count = 0;
             for (let i = 0; i < result.length; i++) {
-                payload.titles.push({ id: result[i].isbn13, label: result[i].title });
-            }
-            sql.pool.query(selectAllConditions, [], (err, result) => {
```

Comment:
This query is still needed for add book modal

Reply:
This too, concerning.